src/components/Filters/FilterDescription.tsx

```diff
@@ -9,14 +9,19 @@ import {
 import { isDateTimeFormat } from '../../DataTypes';
 import { format as dateFormat } from 'date-fns';
 import { isJSONSchema } from '../../AutoUI/schemaOps';
+import isEqual from 'lodash/isEqual';
 
-const transformToRidableValue = (
+const transformToReadableValue = (
 	parsedFilterDescription: SieveFilterDescription,
 ): string => {
 	const { schema, value } = parsedFilterDescription;
 	if (schema && isDateTimeFormat(schema.format)) {
 		return dateFormat(value, 'PPPppp');
 	}
+	if (schema?.enum && 'enumNames' in schema) {
+		const index = schema.enum.findIndex((a) => isEqual(a, value));
+		return (schema.enumNames as string[])[index];
```

Comment:
I guess we could do v
```suggestion
		return String(schema.enumNames[index]);
```